# MHI Bioinformatics Module
## Background
Welcome to the home page of the Microbe Host Interactions bioinformatics module! You'll be able to find everything you need for each class in its respective directory inside this repository.

## About us
### Owen Hale
 - 4th year Biological Sciences PhD candidate in Megan Behringer's lab
 - Uses bioinformatics to study the ecology and evolution of microbes in the lab and inside hosts

### Seth Reasoner
 - 7th year Microbe Host Interactions MD/PhD candidate in Maria Hadjifrangiskou's lab
 - Uses bioinformatics to characterize the microbiome of patients at risk of urinary tract infections

### Jamisha Francis
 - 7th year Microbe Host Interactions PhD candidate in Maria Hadjifrangiskou's lab
 - Uses bioinformatics to characterize the microbiome of the urinary tract

## Module Schedule
 - class 1
 - class 2
 - [Genome Annotation](https://github.com/owenfhale/bioinfo_class/tree/main/genome_annotation)
 - etc.

## Helpful Links
### ACCRE
 - [ACCRE website](https://www.vanderbilt.edu/accre/)
 - [ACCRE cheat sheet](https://cdn.vanderbilt.edu/vu-URL/wp-content/uploads/sites/157/2018/02/19214952/ACCRE-Cheat-Sheet-March-2019.pdf)
### R
 - [R for Data Science](https://r4ds.hadley.nz/)
   
### 16S rRNA Tutorials
 - [Demonstration of 16S rRNA-based microbiome analysis using dada2, phyloseq, LEfSe, picrust2 and other tools
](https://ycl6.github.io/16S-Demo/index.html)
 - [DADA2 Workflow] (https://hypocolypse.github.io/16s-dada2.html)
 - [Processing marker-gene data with DADA2] (https://web.stanford.edu/class/bios221/Pune/Labs/Lab_dada2/Lab_dada2_workflow.html)
