# MHI Bioinformatics Module
## Background
Welcome to the home page of the Microbe Host Interactions bioinformatics module!
You'll be able to find everything you need for each class on this page.

## About the instructors:
### Owen Hale
 - 4th-year Biological Sciences PhD candidate in [Megan Behringer's lab](https://behringerlab.wixsite.com/index)
 - Uses bioinformatics to study ecology and evolution in the human microbiome

### Seth Reasoner
 - 7th-year Microbe Host Interactions MD/PhD candidate in [Maria Hadjifrangiskou's lab](https://my.vanderbilt.edu/hadjifrangiskoulab/)
 - Uses bioinformatics to characterize the microbiome of patients at risk of urinary tract infections

## Module Schedule
1. Introduction to bioinformatics
2. Sequence generation and quality control
3. Genome assembly and annotation
4. Sequence alignment
5. Data visualization with R

***

## Lecture 1: Introduction to bioinformatics
#### Pre-class assignments:
 - Log into [ACCRE account](https://www.vanderbilt.edu/accre/)
 - Complete [ACCRE training](https://www.vanderbilt.edu/accre/required-training/)
#### Lecture slides:
[Click here for lecture slides](https://docs.google.com/presentation/d/1BbIJJxXv6dQ4siVwZKt_75FtPzUPPOCRh7aKnoqrbss/edit?usp=sharing)
#### Post-class assignments:
 - Read [introductory bioinformatics paper](https://doi.org/10.1371/journal.pcbi.1008645)
 - Read [fastp paper](https://doi.org/10.1002/imt2.107)
 - Peruse [fastp documentation](https://github.com/OpenGene/fastp)
 - Install fastp with conda

## Lecture 2: Sequence data generation and quality control
#### Lecture slides:
[Click here for lecture slides](link to slides)
#### Post-class assignments:
 - Read [Unicycler paper](https://doi.org/10.1371/journal.pcbi.1005595)
 - Peruse [Unicycler documentation](https://github.com/rrwick/Unicycler)
 - Read [prokka paper](https://doi.org/10.1093/bioinformatics/btu153)
 - Peruse [prokka documentation](https://github.com/tseemann/prokka)

## Lecture 3: Genome assembly and annotation
#### Lecture slides:
[Click here for lecture slides](link to slides)
#### Post-class assignments:
 - Read [bowtie2 paper](https://doi.org/10.1038/nmeth.1923)
 - Peruse [bowtie2 documentation](https://bowtie-bio.sourceforge.net/bowtie2/index.shtml)

## Lecture 4: Sequence alignment
#### Lecture slides:
[Click here for lecture slides](link to slides)
#### Post-class assignments:
 - Install [R and R Studio](https://posit.co/download/rstudio-desktop/)
 - Read one of the following papers:
     - Short read alignment: [BWA mem]([link to paper](https://doi.org/10.48550/arXiv.1303.3997))
     - Long read/whole-genome alignment: [minimap2](https://doi.org/10.1093/bioinformatics/bty191)
     - Hybrid genome assembly: [Unicycler](https://doi.org/10.1099/mgen.0.000132)
     - Eukaryotic genome annotation: [BRAKER2](https://doi.org/10.1093/nargab/lqaa108)
     - RNA-seq pipeline: [new Tuxedo protocol](https://doi.org/10.1038/nprot.2016.095)
     - TnSeq analysis: [TRANSIT](https://doi.org/10.1371/journal.pcbi.1004401)
     - Multiple sequence alignment: [MAFFT](https://doi.org/10.1093/nar/gkf436)
     - Variant calling: [GATK HaplotypeCaler](https://doi.org/10.1101/201178)
 - Peruse the [ggplot2 documentation](https://ggplot2.tidyverse.org/)
   
## Lecture 5: Data visualization with R
#### Lecture slides:
[Click here for lecture slides](link to slides)
#### Post-class assignment:
 - Work on your final presentation and email your slides to Owen before the next class

## Lecture 6: Your presentations!
Make sure you've sent Owen your slides!

***

## Papers bolstered by genomics
 - Zhu, Meilin, et al. "Vaginal Lactobacillus fatty acid response mechanisms reveal a metabolite-targeted strategy for bacterial vaginosis treatment." Cell (2024).
 - Worthan, Sarah B., et al. "Evolution of pH-sensitive transcription termination in Escherichia coli during adaptation to repeated long-term starvation." Proceedings of the National Academy of Sciences 121.39 (2024): e2405546121.
 - Weiss, Andy, et al. "Zn-regulated GTPase metalloprotein activator 1 modulates vertebrate zinc homeostasis." Cell 185.12 (2022): 2148-2163.

## Helpful Links
### ACCRE
 - [ACCRE website](https://www.vanderbilt.edu/accre/)
 - [ACCRE cheat sheet](https://cdn.vanderbilt.edu/vu-URL/wp-content/uploads/sites/157/2018/02/19214952/ACCRE-Cheat-Sheet-March-2019.pdf)
 - [ACCRE training](https://www.vanderbilt.edu/accre/required-training/)
### R
 - [R for Data Science](https://r4ds.hadley.nz/)
 - [R Graphics Cookbook](https://r-graphics.org/)
### 16S rRNA Tutorials
 - [Demonstration of 16S rRNA-based microbiome analysis using dada2, phyloseq, LEfSe, picrust2 and other tools
](https://ycl6.github.io/16S-Demo/index.html)
 - [DADA2 Workflow](https://hypocolypse.github.io/16s-dada2.html)
 - [Processing marker-gene data with DADA2](https://web.stanford.edu/class/bios221/Pune/Labs/Lab_dada2/Lab_dada2_workflow.html)
 - [DADA2 Pipeline Tutorial](https://benjjneb.github.io/dada2/tutorial_1_2.html)
